lib/librato/metrics/aggregator.rb

```diff
@@ -52,20 +52,29 @@ def initialize(opts={})
       # @return [Aggregator] returns self
       def add(measurements)
         measurements.each do |metric, data|
+          entry = {}
           if @prefix
             metric = "#{@prefix}.#{metric}"
           end
+          entry[:name] = metric.to_s
           if data.respond_to?(:each) # hash form
+            validate_parameters(data)
             value = data[:value]
             if data[:source]
-              metric = "#{metric}#{SOURCE_SEPARATOR}#{data[:source]}"
+              metric = "#{metric}#{SEPARATOR}#{data[:source]}"
+              entry[:source] = data[:source].to_s
+            elsif data[:tags] && data[:tags].respond_to?(:each)
+              metric = Librato::Metrics::Util.build_key_for(metric.to_s, data[:tags])
+              entry[:tags] = data[:tags]
```

Comment:
It looks like the tags are getting serialized into the metric key and stored in the value here as well, is that right?